java/vector/src/test/java/org/apache/arrow/vector/TestStructVector.java

```diff
@@ -69,4 +69,18 @@ public void testMakeTransferPair() {
       assertEquals(0, toChild.getValidityBuffer().capacity());
     }
   }
+
+  @Test
+  public void tesReAlloc() throws Exception {
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("k1", "v1");
+    FieldType type = new FieldType(true, Struct.INSTANCE, null, metadata);
+    try (StructVector vector = new StructVector("struct", allocator, type, null)) {
+      for (int i=0; i<100; ++i) {
```

Comment:
should we not test for the value capacity to assert that it is not expanded? (is that covered here..i was not able to figure out).

Reply:
OK. I will add an assert. Thanks.